google-services/Location/LocationAddress/LocationAddress/MainActivity.cs

```diff
@@ -1,186 +1,292 @@
+
+            mFusedLocationClient = LocationServices.GetFusedLocationProviderClient(this);
+
+            UpdateUiWidgets();
+        }
+
+        protected override void OnStart()
+        {
+            base.OnStart();
+
+            if (!CheckPermissions())
+            {
+                RequestPermissions();
+            }
+            else
+            {
+                GetAddress();
+            }
+        }
+
+        private void UpdateValuesFromBundle(Bundle savedInstanceState)
+        {
+            if (savedInstanceState != null)
+            {
+                if (savedInstanceState.KeySet().Contains(AddressRequestedKey))
+                {
+                    mAddressRequested = savedInstanceState.GetBoolean(AddressRequestedKey);
+                }
+                if (savedInstanceState.KeySet().Contains(LocationAddressKey))
+                {
+                    mAddressOutput = savedInstanceState.GetString(LocationAddressKey);
+                    DisplayAddressOutput();
+                }
+            }
+        }
+
+        public void FetchAddressButtonHandler(object sender, EventArgs eventArgs)
+        {
+            if (mLastLocation != null)
+            {
+                StartIntentService();
+                return;
+            }
+            mAddressRequested = true;
+            UpdateUiWidgets();
+        }
+
+        public void StartIntentService()
+        {
+            Intent intent = new Intent(this, typeof(FetchAddressIntentService));
+            intent.PutExtra(Constants.Receiver, mResultReceiver);
+            intent.PutExtra(Constants.LocationDataExtra, mLastLocation);
+            StartService(intent);
+        }
+
+        public void GetAddress()
+        {
+            var lastLocation = mFusedLocationClient.LastLocation;
+            lastLocation.AddOnSuccessListener(this, new GetAddressOnSuccessListener { Activity = this });
+            lastLocation.AddOnFailureListener(this, new GetAddressOnFailureListener { Activity = this });
+        }
+
+        public void DisplayAddressOutput()
+        {
+            mLocationAddressTextView.Text = mAddressOutput;
+        }
+
+        public void UpdateUiWidgets()
+        {
+            if (mAddressRequested)
+            {
+                mProgressBar.Visibility = ViewStates.Visible;
+                mFetchAddressButton.Enabled = false;
+            }
+            else
+            {
+                mProgressBar.Visibility = ViewStates.Gone;
+                mFetchAddressButton.Enabled = true;
+            }
+        }
+
+        public void ShowToast(String text)
+        {
+            Toast.MakeText(this, text, ToastLength.Short).Show();
+        }
+
+        public void OnSaveInstanceState(Bundle savedInstanceState)
+        {
+            savedInstanceState.PutBoolean(AddressRequestedKey, mAddressRequested);
+            savedInstanceState.PutString(LocationAddressKey, mAddressOutput);
+            OnSaveInstanceState(savedInstanceState);
```

Comment:
Once this method is updated to override the base, you should call into the base implementation here, rather than this presumably infinite recursion which would occur here.

Reply:
Done.